### PR TITLE
Update status handling/unit tests in kfp-ui

### PR DESCRIPTION
**Pull request by ca-scribner**

### CODE REVIEW

File: charms/kfp-ui/tests/unit/test_operator.py [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+    if expected_status is None:
+        assert data == expected_returned_data
+    else:
+        assert partial_relation_data.value.status == expected_status
+
+
+# ingress (receive data)
+
+
+# kfp-ui (provide data)
```
Comment (DomFleischmann) [inline charms/kfp-ui/tests/unit/test_operator.py:205]: Maybe add a `TODO` here.

File: charms/kfp-ui/src/charm.py [hunk 2]
```diff
@@ -253,6 +205,114 @@ def set_pod_spec(self, event):
...
+        # Unpack data (we care only about the first element)
+        data_dict = list(unpacked_relation_data.values())[0]
+
+        # Catch if empty data dict is received (JSONSchema ValidationError above does not raise
+        # when this happens)
+        if len(data_dict) == 0:
+            raise CheckFailedError(
+                f"Found incomplete/incorrect relation data for {relation_name}.",
+                BlockedStatus,
+            )
```
Comment (DomFleischmann) [inline charms/kfp-ui/src/charm.py:301]: Has a bug been filed for this in SDI? If this check remains I would like it to at least have a bug link

**Review by DomFleischmann — APPROVED** [review]
Two small comments

File: charms/kfp-ui/src/charm.py (hunk 2 above)
Comment (ca-scribner) [reply charms/kfp-ui/src/charm.py:301]: https://github.com/canonical/serialized-data-interface/issues/28

**Pull request merged**